### PR TITLE
Make `diff --types` show diff summary as two states

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have added myself to the contributors in `CHANGELOG.md` (optional)

### CODE REVIEW

**Review by arxanas — COMMENTED** [review]
It looks useful for power users, but too difficult to understand for novices to me. With Mercurial, I could immediately guess what A/M/D stand for (in combination with the color cues), but I wouldn't be able to guess what F means or why there's one or two. I kind of wonder if, using the proposed format, I wouldn't be translating it to added/mldified/removed in my head anyways.

I didn't look too carefully; is there a way to express renames or copies in this kind of status output?

**Comment by martinvonz** [comment]
> It looks useful for power users, but too difficult to understand for novices to me. With Mercurial, I could immediately guess what A/M/D stand for (in combination with the color cues), but I wouldn't be able to guess what F means or why there's one or two. I kind of wonder if, using the proposed format, I wouldn't be translating it to added/mldified/removed in my head anyways.

Yes, I'm also quite hesitant about it.

> I didn't look too carefully; is there a way to express renames or copies in this kind of status output?

Should be able to copy Git's way of putting both files on the same line with a `=>` between. They even minimize it like this: `some/{old => new/subdirectory}/path` (from memory)

**Comment by arxanas** [comment]
> Should be able to copy Git's way of putting both files on the same line with a => between. They even minimize it like this: some/{old => new/subdirectory}/path (from memory)

In that case, it seems a little strange to use a "raw" format for the status but then a "refined" format to express the rename. It feels like both should be on the same level of abstraction, somehow.

**Comment by necauqua** [comment]
I think adding this behind a flag (maybe something like --format=somewordforit) into main would be great.
The point is that it basically implements a missing feature (#1111) and I have a usecase for it, I need to know which paths are conflicted in a commit, for the intellij plugin thing :)

File: src/commands.rs [hunk 1]
```diff
@@ -191,7 +191,13 @@ struct PrintArgs {
 #[derive(clap::Args, Clone, Debug)]
 #[command(group(ArgGroup::new("format").args(&["summary", "git", "color_words"])))]
 struct DiffFormatArgs {
-    /// For each path, show only whether it was modified, added, or removed
+    /// For each path, show only its type before and after
+    ///
+    /// The diff is shows as two letters. The first letter indicates the type
```
Comment (isinyaaa) [inline src/commands.rs:196]: nit: typo

```suggestion
    /// The diff is shown as two letters. The first letter indicates the type
```

**Comment by martinvonz** [comment]
> I quite like the summary format the way you did it, but @necauqua's idea seems better for other use cases, maybe having a way to specify the default behavior in `.jjconfig.toml` would be a good idea too.
> 
> Something like:
> 
> ```toml
> log.summary.format = {traditional,fileTypeDiff,verbose}
> ```
> 
> Where `fileTypeDiff` would be the way you propose on this MR.

I only added a `--types` flag for now since that's what @necauqua needed, but I think your suggestion makes sense.

**Review by yuja — APPROVED** [review]
> log.summary.format = {traditional,fileTypeDiff,verbose}

Perhaps, we'll need separate config options for short/long format?
`diff.long-format = git|color-words`, `diff.short-format = summary|types`

**Pull request merged**